src/components/dashboard/main-content/course-enrollments/CourseEnrollments.jsx

```diff
@@ -53,12 +57,12 @@ const CourseEnrollments = ({ children }) => {
     );
     setAssignments(assignmentsData);
```

Comment:
[suggestion] I might propose a bit of a further refactor to this `useEffect` to clean it up a bit:

```jsx
useEffect(() => {
    if (!redeemableLearnerCreditPolicies) {
      return;
    }

    const {
      allocatedAssignments,
      canceledAssignments,
      hasCanceledAssignments,
    } = redeemableLearnerCreditPolicies.learnerContentAssignments;

    const sortedAllocatedAssignments = sortAssignmentsByAssignmentStatus(allocatedAssignments);
    const transformedAllocatedAssignments = getTransformedAllocatedAssignments(
      sortedAllocatedAssignments,
      slug,
    );
    setAssignments(transformedAllocatedAssignments);

    const hasActiveCancelledAssignments = (
      hasCanceledAssignments && getHasActiveCancelledAssignments(canceledAssignments)
    );
    setShowCancelledAssignmentsAlert(hasActiveCancelledAssignments);

    const hasActiveExpiredAssignments = getHasActiveExpiredAssignment(allocatedAssignments);
    setShowExpiredAssignmentsAlert(hasActiveExpiredAssignments);
}, [redeemableLearnerCreditPolicies, slug]);
```

By doing so, we could rely on the state variable `assignments` instead of `assignedCourses` when passing them as the course runs to the assigned `CourseSection` component (i.e., the call to `getTransformedAllocatedAssignments` was moved within the `useEffect`).